src/profiler.cpp

```diff
@@ -995,8 +995,11 @@ Engine* Profiler::selectEngine(const char* event_name) {
 
 Engine* Profiler::selectAllocEngine(long alloc_interval, bool live) {
     if (VM::canSampleObjects()) {
-        return &object_sampler;
-    } else if (VM::isOpenJ9()) {
+        if (VM::addCanSampleObjectsCapability() == 0) {
```

Comment:
Does it really need to be under `canSampleObjects`? Can we just try acquire capability and if it returns an error, fall back to a different engine.

Reply:
will remove canSampleObjects